src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.DataConnection.cs

```diff
@@ -87,7 +85,7 @@ public bool AllowAdd
                 if (CurrencyManager is not null)
                 {
                     // we only allow to add new rows on an IBindingList
-                    return (CurrencyManager.List is IBindingList) && CurrencyManager.AllowAdd && ((IBindingList)CurrencyManager.List).SupportsChangeNotification;
+                    return (CurrencyManager.List is IBindingList bindingList) && CurrencyManager.AllowAdd && bindingList.SupportsChangeNotification;
```

Comment:
```suggestion
                    return CurrencyManager is { List: IBindingList { SupportsChangeNotification: true }, AllowAdd: true };
```
and that could even replace the null check of `CurrencyManager`

Reply:
I can't say that I am a big fan of this syntax. Let's see what the maintainers think.

Reply:
I understand your point, and this is basically just a nit. I just thought in this case it was nice to forego a variable declaration entirely

Reply:
@gpetrou and @halgab, please see my comment [here ](https://github.com/dotnet/winforms/pull/9966#discussion_r1341602147)